pkg/kubelet/kubelet.go

```diff
@@ -1898,6 +1900,23 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		}
 	}
 
+	// If pod-level resources are set, CPU and Memory managers alignment is skipped,
+	// an event is surfaced to inform the user when skipped in the corresponding pod.
+	if updateType == kubetypes.SyncPodCreate &&
+		utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) &&
+		resourcehelper.IsPodLevelResourcesSet(pod) &&
+		v1qos.GetPodQOS(pod) == v1.PodQOSGuaranteed {
+
+		if kl.containerManager.GetNodeConfig().CPUManagerPolicy == string(cpumanager.PolicyStatic) {
```

Comment:
please no. This is another instance (alongside in-place VPA) calls for https://github.com/kubernetes/kubernetes/pull/128728

Reply:
I made some changes that allow to produce the event directly from the CPU and Memory managers, instead of producing it from the Kubelet. What do you think?

Reply:
I think you approach is better when dealing with other modules outside the managers and for the overall policies allocation. However, with this PR I am looking to produce an event directly from the managers when preventing exclusive allocation and hint generation for the Topology manager in regards to the specific pods (when they make use of pod level resources). 

Anyway, I think your PR could be leveraged to also consider hint generation and add specific validation for the pods.

Reply:
Thanks for elaborating. If my PR doesn't merge quickly (and it seems likely it will NOT merge quickly... :\ ) I'm fine exploring alternative approaches and perhaps reconcile/refactor later on.